sklearnex/ensemble/forest.py

```diff
@@ -16,3 +16,1146 @@
+            'min_weight_fraction_leaf': self.min_weight_fraction_leaf,
+            'max_features': self.max_features,
+            'max_leaf_nodes': self.max_leaf_nodes,
+            'min_impurity_decrease': self.min_impurity_decrease,
+            'random_state': None,
+        }
+        if not sklearn_check_version('1.0'):
+            params['min_impurity_split'] = self.min_impurity_split
+        est = ExtraTreeClassifier(**params)
+        # we need to set est.tree_ field with Trees constructed from Intel(R)
+        # oneAPI Data Analytics Library solution
+        estimators_ = []
+        random_state_checked = check_random_state(self.random_state)
+        for i in range(self.n_estimators):
+            est_i = clone(est)
+            est_i.set_params(
+                random_state=random_state_checked.randint(
+                    np.iinfo(
+                        np.int32).max))
+            if sklearn_check_version('1.0'):
+                est_i.n_features_in_ = self.n_features_in_
+            else:
+                est_i.n_features_ = self.n_features_in_
+            est_i.n_outputs_ = self.n_outputs_
+            est_i.classes_ = classes_
+            est_i.n_classes_ = n_classes_
+            tree_i_state_class = get_tree_state_cls(
+                self._onedal_model, i, n_classes_)
+            tree_i_state_dict = {
+                'max_depth': tree_i_state_class.max_depth,
+                'node_count': tree_i_state_class.node_count,
+                'nodes': tree_i_state_class.node_ar,
+                'values': tree_i_state_class.value_ar}
+            est_i.tree_ = Tree(
+                self.n_features_in_,
+                np.array(
+                    [n_classes_],
+                    dtype=np.intp),
+                self.n_outputs_)
+            est_i.tree_.__setstate__(tree_i_state_dict)
+            estimators_.append(est_i)
+
+        self._cached_estimators_ = estimators_
+        return estimators_
+
+    def _onedal_cpu_supported(self, method_name, *data):
+        if method_name == 'ensemble.ExtraTreesClassifier.fit':
+            ready, X, y, sample_weight = self._onedal_ready(*data)
+            if self.splitter_mode == 'random':
+                warnings.warn("'random' splitter mode supports GPU devices only "
+                              "and requires oneDAL version >= 2023.1.1. "
+                              "Using 'best' mode instead.", RuntimeWarning)
+                self.splitter_mode = 'best'
+            if not ready:
+                return False
+            elif sp.issparse(X):
+                return False
+            elif sp.issparse(y):
+                return False
+            elif sp.issparse(sample_weight):
+                return False
+            elif not self.ccp_alpha == 0.0:
+                return False
+            elif self.warm_start:
+                return False
+            elif self.oob_score and not daal_check_version((2023, 'P', 101)):
```

Comment:
If oneDAL ExtraTrees were implemented in 2023.2.0, change to:
```suggestion
            elif self.oob_score and not daal_check_version((2023, 'P', 200)):
```

Reply:
If ExtraTrees is forced to use splitter_mode='best' then it will act similarly to Random Forest, and then would need to go through checks like random forest. You, Samir, and I should talk about how to resolve public facing API issues, esp. as it relates to preview/forest.py and this extratrees.